### PR TITLE
Adds changes for using custom launch template

### PR DESCRIPTION
**Pull request by ashishranjan738**
This commit adds changes for using custom launch template for worker nodes to PVRE and IMDSV2 issues.

Issue #, if available:

Description of changes:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by hakuna-matatah** [comment]
Is this tested internally ? Have we seen successful test run ?

File: tests/tasks/setup/eks/awscli-cfn-lt.yaml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      description: The region where the cluster is in.
+    - name: kubernetes-version
+      default: "1.28"
+      description: The EKS version to install.
+    - name: ng-cfn-url
+      description: The url of the CFN YAML/JSON to create CFN stack for NG launch template
+    - name: endpoint
+      default: ""
+    - name: node-max-pod-count
+      default: "35"
```
Comment (hakuna-matatah) [inline tests/tasks/setup/eks/awscli-cfn-lt.yaml:27]: Why 35 ? As long as we use PD, we should be able to allocate more than 35 ?

**Comment by ashishranjan738** [comment]
> Is this tested internally ? Have we seen successful test run ?

yes

File: tests/tasks/setup/eks/awscli-cfn-lt.yaml (hunk 1 above)
Comment (ashishranjan738) [reply tests/tasks/setup/eks/awscli-cfn-lt.yaml]: We don't need this anymore

**Review by hakuna-matatah — APPROVED** [review]

**Pull request merged**